hudi-common/src/main/java/org/apache/hudi/common/util/CleanerUtils.java

```diff
@@ -168,10 +168,18 @@ public static Option<HoodieInstant> getEarliestCommitToRetain(
   public static HoodieCleanerPlan getCleanerPlan(HoodieTableMetaClient metaClient, HoodieInstant cleanInstant)
       throws IOException {
     CleanPlanMigrator cleanPlanMigrator = new CleanPlanMigrator(metaClient);
+    cleanInstant = getCleanRequestInstant(metaClient, cleanInstant);
     HoodieCleanerPlan cleanerPlan = metaClient.getActiveTimeline().readCleanerPlan(cleanInstant);
     return cleanPlanMigrator.upgradeToLatest(cleanerPlan, cleanerPlan.getVersion());
   }
 
+  public static HoodieInstant getCleanRequestInstant(HoodieTableMetaClient metaClient, HoodieInstant cleanInstant) {
+    if (cleanInstant.isInflight() || cleanInstant.isCompleted()) {
```

Comment:
we can check if it is not `REQUESTED` instead.

Reply:
> we can check if it is not `REQUESTED` instead.

Done~